### PR TITLE
Added system test pre-stage jars

### PR DESCRIPTION
**Pull request by annaibm**
- Enables to avoid system test prerequisite zip file being empty.

related: https://github.com/adoptium/aqa-tests/issues/4912

### CODE REVIEW

File: scripts/getDependencies.pl [hunk 1]
```diff
@@ -175,14 +175,68 @@
...
+	},
+	log4j_core => {
+		url => 'https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-core/2.15.0/log4j-core-2.15.0.jar',
+		fname => 'log4j-core.jar',
+		sha1 => 'ba55c13d7ac2fd44df9cc8074455719a33f375b9'
+	},
+	mauve => {
+		url => 'https://ci.adoptium.net/job/systemtest.getDependency/lastSuccessfulBuild/artifact/systemtest_prereqs/mauve/mauve.jar',
+		fname => 'mauve.jar',
+		sha1 => '8ed5b172be6a8885b72d0015f44e2a0b6c172d47'
```
[the remark this answers is not in the document]
Comment (llxia) [reply scripts/getDependencies.pl:238]: This has not been addressed. 

We should not ignore checksum for all systemtests jars. It means  line 329 `my $ignoreChecksum = ($url =~ /systemtest.getDependency/);` is wrong. Using `mauve` as an example, it should not have `sha1`. And we should update the logic to if `sha1` or `shaurl` does not exist, ignore checksum.

[the remark this answers is not in the document]
Comment (annaibm) [reply scripts/getDependencies.pl]: @llxia I have updated the line to address the logic to if sha1 or shaurl does not exist, ignore checksum as below.
`my $ignoreChecksum = (!defined $sha1 || $sha1 eq '') && (!defined $shaurl || $shaurl eq ''); `.

File: scripts/getDependencies.pl [hunk 2]
```diff
@@ -242,15 +303,31 @@
...
 		my $url = $jars_info[$i]{url};
 		my $fn = $jars_info[$i]{fname};
-		my $filename = $path . $sep . $fn;
+		my $sha1 = $jars_info[$i]{sha1};
+		my $dir = $jars_info[$i]{dir} // "";
+		my $full_dir_path = File::Spec->catdir($path, $dir);
+		my $url_custom = $customUrl;
+
+		if (!-d $full_dir_path) {
+			make_path($full_dir_path, {chmod => 0777, verbose => 1}) or die "Failed to create directory: $full_dir_path: $!";
```
Comment (llxia) [inline scripts/getDependencies.pl:312]: change `chmod => 0777` to chmod => `0755`

**Comment by annaibm** [comment]
Grinder links:
https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40931/

**Comment by llxia** [comment]
Note: pre-stage system test jars in a separate folder are a temporary solution. Eventually, all jars will be moved into the same dir and duplicate jars will be removed. See https://github.com/adoptium/aqa-tests/issues/4912#issuecomment-2064199172

**Comment by llxia** [comment]
Noticed in the [job console](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/40912/consoleFull) that SysinternalsSuite.zip is downloaded at the runtimes and moved into `testDependency\system_lib\windows_sysinternals`. This is NOT due to this PR as SysinternalsSuite.zip is not even in the `systemtest.getDependency` job. However, we should handle this as well. I will open a separate PR for this.

```
00:08:47.503  configure-windows-sysinternals:
00:08:47.503       [echo] Executing macro download-file
00:08:47.503       [echo] File to download: https://download.sysinternals.com/files/SysinternalsSuite.zip
00:08:47.503       [echo] Destination: C:\temp\/windows_sysinternals/SysinternalsSuite.zip
00:08:47.503       [echo] Download tool: curl
00:08:47.503      [mkdir] Created dir: C:\temp\windows_sysinternals
00:08:47.503       [exec]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
00:08:47.503       [exec]                                  Dload  Upload   Total   Spent    Left  Speed
00:08:47.503       [exec] 
00:08:48.583       [exec]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0 95 50.7M   95 48.5M    0     0  38.7M      0  0:00:01  0:00:01 --:--:-- 39.0M100 50.7M  100 50.7M    0     0  39.3M      0  0:00:01  0:00:01 --:--:-- 39.6M
00:08:48.583      [unzip] Expanding: C:\temp\windows_sysinternals\SysinternalsSuite.zip into C:\Users\jenkins\testDependency\system_lib\windows_sysinternals
00:09:12.595     [delete] Deleting directory C:\temp\windows_sysinternals
```

**Review by llxia — APPROVED** [review]
LGTM

**Review by LongyuZhang — APPROVED** [review]
LGTM

**Pull request merged**